### PR TITLE
[azure] create multi clusters in same subnets

### PR DESCRIPTION
**Pull request by jinyunma**
Auto case: OCP-27002 [IPI-on-Azure] create multiple clusters using the same subnets from an exisiting VNet

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28

**Comment by jinyunma** [comment]
Tested job before last [commit](https://github.com/openshift/release/compare/449a4c48ad7d01507216e41dbe206b5c8482abdd..f25a96c7cd6ae1727afabc511ea602299e4958c3): https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58310/rehearse-58310-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28/1851793641954086912

cc @jianlinliu @gpei to review

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/multi-clusters/same-subnets/provision/cucushift-installer-rehearse-azure-cases-multi-clusters-same-subnets-provision-commands.sh [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+function scale_up_worker_nodes()
+{
+    local machineset_name replicas total_nodes_count ret=0
+    total_nodes_count=$(oc get node --no-headers | grep worker | wc -l)
+    machineset_name=$(oc -n openshift-machine-api get -o "jsonpath={range .items[*]}{.metadata.name}{'\\n'}{end}" machinesets | grep worker | head -n1)
+    replicas=$(oc get machineset ${machineset_name} -n openshift-machine-api -o "jsonpath={.spec.replicas}")
+
+    echo "scale up worker on machineset ${machineset_name}..."
+    timeout 5m oc scale --replicas=$((replicas+1)) machineset ${machineset_name} -n openshift-machine-api || ret=1
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/multi-clusters/same-subnets/provision/cucushift-installer-rehearse-azure-cases-multi-clusters-same-subnets-provision-commands.sh:44]: I guess we can merge target replicas and total_nodes_count as one variable.

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/multi-clusters/same-subnets/provision/cucushift-installer-rehearse-azure-cases-multi-clusters-same-subnets-provision-commands.sh:44]: there will be several machinesets created if installing cluster in region with zone supported ( 1 machineset per zone). In this case, replicas does not equal to total_nodes_count.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/cases/multi-clusters/same-subnets/provision/cucushift-installer-rehearse-azure-cases-multi-clusters-same-subnets-provision-commands.sh:44]: Ah, you are right, I forgot it.

**Comment by jinyunma** [comment]
[job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58310/rehearse-58310-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-azure-ipi-multi-clusters-same-subnets-arm-f28/1851848713371127808) passed.

 /pj-rehearse ack

**Pull request merged**